core/network/nic.go

```diff
@@ -210,6 +210,32 @@ func (i *InterfaceInfo) IsVLAN() bool {
 	return i.VLANTag > 0
 }
 
+// Validate checks that the receiver looks like a real interface.
+// An error is returned if invalid members are detected.
+func (i *InterfaceInfo) Validate() error {
+	if i.MACAddress != "" {
+		if _, err := net.ParseMAC(i.MACAddress); err != nil {
+			return errors.NotValidf("link-layer device hardware address %q", i.MACAddress)
+		}
+	}
+
+	if i.InterfaceName == "" {
+		return errors.NotValidf("link-layer device %q, empty name", i.MACAddress)
+	}
+
+	if !IsValidLinkLayerDeviceName(i.InterfaceName) {
+		// TODO (manadart 2020-07-07): This preserves prior behaviour.
+		// If we are waving invalid names through, I'm not sure of the value.
+		logger.Warningf("link-layer device %q has an invalid name, %q", i.MACAddress, i.InterfaceName)
```

Comment:
What's the side-effect of passing a bogus name here? Just an entry in the DB or does it affect any of the workers?

Reply:
Just an entry in the database. These checks are a subset of those done inside the state package against `LinkLayerDeviceArgs`, which of course is too deep for this to happen.